src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.Core.cs

```diff
@@ -78,24 +78,23 @@ private bool InitializeBinaryFormatter()
             // If BinaryFormatter support is disabled for the app, the linker will replace this entire
             // method body with "return false;", skipping all reflection code below.
 
-            LazyInitializer.EnsureInitialized(ref s_binaryFormatterType, static () =>
-                Type.GetType("System.Runtime.Serialization.Formatters.Binary.BinaryFormatter, System.Runtime.Serialization.Formatters, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
-                throwOnError: true)!);
+            Interlocked.CompareExchange(
+                ref s_binaryFormatterType,
+                Type.GetType("System.Runtime.Serialization.Formatters.Binary.BinaryFormatter, System.Runtime.Serialization.Formatters, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: true),
```

Comment:
Nit: The fully qualified type name can be just "System.Runtime.Serialization.Formatters.Binary.BinaryFormatter, System.Runtime.Serialization.Formatters". The strong name, etc. is ignored by the .NET Core loader.